### PR TITLE
[Kotlin Migration] CsvReaderIterator 

### PR DESCRIPTION
**Pull request by codingtosh**
[Kotlin Migration] CsvReaderIterator

### CODE REVIEW

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
`@KotlinCleanup("ensure code style matches https://github.com/python/cpython/blob/c88239f864a27f673c0f0a9e62d2488563f9d081/Modules/_csv.c")`

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/python/CsvReader.kt [hunk 1]
```diff
@@ -34,7 +34,10 @@ class CsvReader(data: Iterator<String>, delimiter: Char, inputDialect: CsvDialec
         if (iter == null) {
             iter = CsvReaderIterator(this)
         }
-        return iter!!
+
+        @Suppress("UNCHECKED_CAST")
+        val _iter = iter as MutableIterator<List<String>>?
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/python/CsvReader.kt:39]: I don't think this is correct. 

`MutableIterator<List<String>?>`, not `MutableIterator<List<String>>?`. This implies the code signature of the class is incorrect.

This code is in use in one of my draft PRs, which may be used to test

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/python/CsvReaderIterator.kt [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        numeric_field = 0
+    }
+
+    // Copied from C code
+    private fun parse_process_char(_c: Char): Int {
+        var c = _c
+        val dialect = reader.dialect
+        when (state) {
+            State.START_RECORD -> {
+                /* start of record */if (c == '\u0000') { /* empty line - return [] */
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/python/CsvReaderIterator.kt:93]: Issues with spacing here. It should match the `c` file as much as possible, but this looks incorrect enough to fix in the PR as it's just spacing.

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
This wasn't a fun class to write, and won't be a fun class to convert

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/python/CsvReaderIterator.kt [hunk 3]
```diff
@@ -18,69 +18,39 @@
...
-        START_FIELD,
-        IN_QUOTED_FIELD,
-        EAT_CRNL,
-        AFTER_ESCAPED_CRNL,
-        ESCAPED_CHAR,
-        IN_FIELD,
-        ESCAPE_IN_QUOTED_FIELD,
-        QUOTE_IN_QUOTED_FIELD
+    internal enum class State {
+        START_RECORD, START_FIELD, IN_QUOTED_FIELD, EAT_CRNL, AFTER_ESCAPED_CRNL, ESCAPED_CHAR, IN_FIELD, ESCAPE_IN_QUOTED_FIELD, QUOTE_IN_QUOTED_FIELD
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/python/CsvReaderIterator.kt:45]: If this is not too much work, I'd love to see the list with one element by line.
Not only it's currently a long line, but it'd also help the diff displaying correctly

Comment (codingtosh) [reply AnkiDroid/src/main/java/com/ichi2/libanki/importer/python/CsvReaderIterator.kt]: Sure, done. May I also suggest that we enable trailing commas in the project codestyle settings?

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/python/CsvReaderIterator.kt (hunk 2 above)
Comment (codingtosh) [reply AnkiDroid/src/main/java/com/ichi2/libanki/importer/python/CsvReaderIterator.kt:102]: I'm running lint checks
`./gradlew ktlintcheck --info`
and there are issues with indentation and newline if I try to make it similar to `_csv.c`

**Pull request closed without merging**